ygot/render.go

```diff
@@ -1370,11 +1372,11 @@ func jsonValue(field reflect.Value, parentMod string, args jsonOutputConfig) (in
 		var err error
 		switch {
 		case util.IsValueInterfaceToStructPtr(field):
-			if value, err = unionInterfaceValue(field, appmod); err != nil {
+			if value, err = unwrapUnionInterfaceValue(field, appmod); err != nil {
 				return nil, err
 			}
-		case field.Elem().Kind() == reflect.Ptr && field.Elem().Elem().Type().Name() == BinaryTypeName && field.Elem().Elem().Kind() == reflect.Slice:
-			if value, err = jsonSlice(field.Elem().Elem(), parentMod, args); err != nil {
+		case field.Elem().Kind() == reflect.Slice && field.Elem().Type().Name() == BinaryTypeName:
```

Comment:
Can you explain this change please?

Previously, we had an interface was dereferenced by `Elem` to get to a pointer which was dereferenced by `Elem` to get to the slice, which we then checked whether it was a `Binary` or not. In this changed code I don't see handling for that case.

In the changed version it looks like you just handle the new `union` types, such that we can't now handle a `union` that contains a `binary` of the old format.

Looking at  https://github.com/openconfig/ygot/blob/master/ygot/render_test.go, we're missing coverage for the old format of `union` that contains `binary`, so it might be worth adding here.

Reply:
The change you pointed out is after `Binary` is changed to use value receiver to satisfy the new union API instead of using pointer receiver. Now `*Binary` is no longer accepted.

Binary tests and fixes added for wrapper unions.

Reply:
Oh, I'm sorry - this diff is not from HEAD, it's a diff within this version. This makes more sense.

My understanding was that we had not added this `field.Elem().Kind()` check in this branch, but had rather added it previously, and we were now changing it. This makes more sense to me.

Thanks for adding the test coverage!